ludwig/api.py

```diff
@@ -878,15 +878,12 @@ def evaluate(
 
             # calculate the overall metrics
             if collect_overall_stats:
-                # TODO ray: support calculating stats on partitioned datasets
                 if self.backend.df_engine.partitioned:
-                    raise ValueError(
-                        "Cannot calculate overall stats on a partitioned DataFrame at this time. "
-                        "Set `calculate_overall_stats=False`."
-                    )
+                    # If using a partitioned engine, we need to convert the RayDataset into a Dask dataframe.
+                    dataset = dataset.ds.to_dask()
```

Comment:
Reaching into the `ds` property here violates the interface (you could imagine in the future supporting other types of partitioned backends in the future other than RayDataset).

I would suggest instead we do something like have the `Dataset` base class provide a `to_df()` method or similar we can call, which would also allow us to obtain a Pandas DF directly from the `PandasDataset`.